### PR TITLE
[lld][WebAssembly] Implement `--start-lib`/`--end-lib`

### LINKED ISSUE #77960: Add support for --start-lib / --end-lib to Wasm LLD to omit the use of archives
Request implementing `-start-lib` and `-end-lib` flags for `wasm-ld`, analogous to the flags in `ld.lld`.

Object files after `-start-lib` are included in the link only when needed to resolve undefined symbols. (`-end-lib` returns command-line processing to the original behavior)

This mimics static libraries without needing to create an archive.

+cc: @sbc100 @walkingeyerobot @googlewalt

### PR DESCRIPTION
**Pull request by sbc100**
Fixes: #77960

### CODE REVIEW

File: lld/wasm/Driver.cpp [hunk 1]
```diff
@@ -94,6 +94,9 @@ class LinkerDriver {
   // True if we are in --whole-archive and --no-whole-archive.
   bool inWholeArchive = false;
 
+  // True if we are in --start-lib and --end-lib.
```
Comment (dschuff) [inline lld/wasm/Driver.cpp:97]:
```suggestion
  // True if we are between --start-lib and --end-lib.
```

File: lld/wasm/Driver.cpp [hunk 2]
```diff
@@ -375,6 +378,16 @@ void LinkerDriver::createFiles(opt::InputArgList &args) {
...
       inWholeArchive = false;
       break;
+    case OPT_start_lib:
+      if (inLib)
+        error("nested --start-lib");
+      inLib = true;
+      break;
+    case OPT_end_lib:
+      if (!inLib)
+        error("stray --end-lib");
```
Comment (dschuff) [inline lld/wasm/Driver.cpp:388]:
```suggestion
        error("--end-lib without matching --start-lib");
```

**Review by dschuff — APPROVED** [review]

File: lld/wasm/Driver.cpp (hunk 2 above)
Comment (sbc100) [reply lld/wasm/Driver.cpp:388]: These match the ELF errors, so maybe we should leave them the same for now?  And maybe followup by making them more clear?

File: lld/wasm/Driver.cpp (hunk 1 above)
Comment (sbc100) [reply lld/wasm/Driver.cpp:97]: Just matching the comment above and in the ELF linker here.  I'm not sure this is a huge grammatical error.

File: lld/wasm/Driver.cpp (hunk 2 above)
Comment (dschuff) [reply lld/wasm/Driver.cpp:388]: Yeah, that's fine. Too bad about all this duplication :)

**Pull request merged**